### ISSUE TITLE
Error updating packages: `*.tar.gz: Cannot open: No such file or directory`

### ISSUE THREAD

**Issue by isaactpetersen** [rstudio/renv#953]
I'm trying to use `{renv}` in a repo on a linux cluster. I receive an error when I try to update packages using `renv::update()`. I confirmed that I cannot install packages using `renv::install()`, either, at least with the packages listed below. It used to work for me in the previous version of `{renv}`, so I'm not sure what changed. Because I'm on a cluster, I cannot update the R version. I am using the latest version of `{renv}`.

Here's the error I receive:
```
> renv::update()
* Querying repositories for available source packages ... Done!
* Checking for updated packages ... Done!
The following package(s) will be updated:

# CRAN ===============================
- DT           [0.20 -> 0.21]
- admisc       [0.24 -> 0.25]
- clipr        [0.7.1 -> 0.8.0]
- colorspace   [2.0-2 -> 2.0-3]
- evaluate     [0.14 -> 0.15]
- future       [1.23.0 -> 1.24.0]
- glue         [1.6.1 -> 1.6.2]
- jsonlite     [1.7.3 -> 1.8.0]
- mgcv         [1.8-38 -> 1.8-39]
- recipes      [0.1.17 -> 0.2.0]
- tidyselect   [1.1.1 -> 1.1.2]
- yaml         [2.2.2 -> 2.3.5]

Do you want to proceed? [y/N]: y
Retrieving 'https://mirror.las.iastate.edu/CRAN/src/contrib/DT_0.21.tar.gz' ...
        OK [file is up to date]
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar xf '~/.local/share/renv/source/repository/DT/DT_0.21.tar.gz' -C '/localscratch/RtmpbcDR0E/renv-description-35648464ce78e' 'DT/DESCRIPTION'
================================================================================

/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: ~/.local/share/renv/source/repository/DT/DT_0.21.tar.gz: Cannot open: No such file or directory
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: Error is not recoverable: exiting now
Error: error decompressing archive [error code 2]
Traceback (most recent calls last):
18: renv::update()
17: install(packages = updates, library = libpaths, rebuild = rebuild,
        project = project)
16: retrieve(names(remotes))
15: handler(package, renv_retrieve_impl(package))
14: renv_retrieve_impl(package)
13: renv_retrieve_repos(record)
12: renv_retrieve_repos_impl(record)
11: renv_retrieve_package(record, url, path)
10: renv_retrieve_successful(record, path)
 9: renv_description_read(path, subdir = subdir)
 8: filebacked(scope = "DESCRIPTION", path = path, callback = renv_description_read_impl,
        subdir = subdir, ...)
 7: callback(path, ...)
 6: renv_archive_decompress(path, files = file, exdir = exdir)
 5: renv_archive_decompress_tar(archive, files = files, exdir = exdir,
        ...)
 4: renv_tar_decompress(tar, archive = archive, files = files, exdir = exdir,
        ...)
 3: renv_system_exec(tar, args, action = "decompressing archive")
 2: stopf(fmt, action, status)
 1: stop(sprintf(fmt, ...), call. = call.)
```

Here's another example:
```
> renv::install("admisc")
Retrieving 'https://mirror.las.iastate.edu/CRAN/src/contrib/admisc_0.25.tar.gz' ...
        OK [file is up to date]
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar xf '~/.local/share/renv/source/repository/admisc/admisc_0.25.tar.gz' -C '/localscratch/RtmpbcDR0E/renv-description-3564867f774cd' 'admisc/DESCRIPTION'
================================================================================

/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: ~/.local/share/renv/source/repository/admisc/admisc_0.25.tar.gz: Cannot open: No such file or directory
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: Error is not recoverable: exiting now
Error: error decompressing archive [error code 2]
Traceback (most recent calls last):
17: renv::install("admisc")
16: retrieve(names(remotes))
15: handler(package, renv_retrieve_impl(package))
14: renv_retrieve_impl(package)
13: renv_retrieve_repos(record)
12: renv_retrieve_repos_impl(record)
11: renv_retrieve_package(record, url, path)
10: renv_retrieve_successful(record, path)
 9: renv_description_read(path, subdir = subdir)
 8: filebacked(scope = "DESCRIPTION", path = path, callback = renv_description_read_impl,
        subdir = subdir, ...)
 7: callback(path, ...)
 6: renv_archive_decompress(path, files = file, exdir = exdir)
 5: renv_archive_decompress_tar(archive, files = files, exdir = exdir,
        ...)
 4: renv_tar_decompress(tar, archive = archive, files = files, exdir = exdir,
        ...)
 3: renv_system_exec(tar, args, action = "decompressing archive")
 2: stopf(fmt, action, status)
 1: stop(sprintf(fmt, ...), call. = call.)
```

Here's my `sessionInfo()`:
```
> sessionInfo()
R version 4.0.5 (2021-03-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: CentOS Linux 7 (Core)

Matrix products: default
BLAS/LAPACK: /opt/ssoft/apps/2021.1/linux-centos7-broadwell/gcc-9.3.0/intel-mkl-2020.4.304-lijseyq/compilers_and_libraries_2020.4.304/linux/mkl/lib/intel64_lin/libmkl_gf_lp64.so

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C
 [9] LC_ADDRESS=C               LC_TELEPHONE=C
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base

loaded via a namespace (and not attached):
[1] compiler_4.0.5      BiocManager_1.30.16 tools_4.0.5
[4] renv_0.15.3
```

**Comment by kevinushey** [comment]
Thanks for the bug report -- this looks like a regression in `renv`. For now, I'd recommend sticking with `renv 0.15.2`. I'll take a look soon.

**Comment by kevinushey** [comment]
You can also try setting:

```
options(renv.userdir.override = path.expand("~/.local/share/renv"))
```

before calling the associated `renv` APIs.

**Comment by isaactpetersen** [comment]
Thanks for looking into into this. I'm still getting an error when running that command before `renv::update()`:

```
> options(renv.userdir.override = path.expand("~/.local/share/renv"))
> renv::update()
* Querying repositories for available source packages ... Done!
* Checking for updated packages ... Done!
The following package(s) will be updated:

# CRAN ===============================
- DT           [0.20 -> 0.21]
- admisc       [0.24 -> 0.25]
- clipr        [0.7.1 -> 0.8.0]
- colorspace   [2.0-2 -> 2.0-3]
- evaluate     [0.14 -> 0.15]
- future       [1.23.0 -> 1.24.0]
- glue         [1.6.1 -> 1.6.2]
- jsonlite     [1.7.3 -> 1.8.0]
- mgcv         [1.8-38 -> 1.8-39]
- recipes      [0.1.17 -> 0.2.0]
- tidyselect   [1.1.1 -> 1.1.2]
- yaml         [2.2.2 -> 2.3.5]

Do you want to proceed? [y/N]: y
Retrieving 'https://mirror.las.iastate.edu/CRAN/src/contrib/DT_0.21.tar.gz' ...
        OK [file is up to date]
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar xf '~/.local/share/renv/source/repository/DT/DT_0.21.tar.gz' -C '/loca                                                                                                                lscratch/RtmpHQYWUh/renv-description-2f9dc2b22c937' 'DT/DESCRIPTION'
================================================================================

/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: ~/.local/share/renv/source/repository/DT/DT_0.21.tar.gz: Cannot open:                                                                                                                 No such file or directory
/usr/share/apps/2021.1/arch/gcc-9.3.0/tar-1.34/bin/tar: Error is not recoverable: exiting now
Error: error decompressing archive [error code 2]
Traceback (most recent calls last):
18: renv::update()
17: install(packages = updates, library = libpaths, rebuild = rebuild,
        project = project)
16: retrieve(names(remotes))
15: handler(package, renv_retrieve_impl(package))
14: renv_retrieve_impl(package)
13: renv_retrieve_repos(record)
12: renv_retrieve_repos_impl(record)
11: renv_retrieve_package(record, url, path)
10: renv_retrieve_successful(record, path)
 9: renv_description_read(path, subdir = subdir)
 8: filebacked(scope = "DESCRIPTION", path = path, callback = renv_description_read_impl,
        subdir = subdir, ...)
 7: callback(path, ...)
 6: renv_archive_decompress(path, files = file, exdir = exdir)
 5: renv_archive_decompress_tar(archive, files = files, exdir = exdir,
        ...)
 4: renv_tar_decompress(tar, archive = archive, files = files, exdir = exdir,
        ...)
 3: renv_system_exec(tar, args, action = "decompressing archive")
 2: stopf(fmt, action, status)
 1: stop(sprintf(fmt, ...), call. = call.)
```

**Comment by kevinushey** [comment]
Thanks for taking the time to test. I think this should now be properly resolved in the development version of `renv`; are you able to test with e.g.

```
renv::install("rstudio/renv")
```

and then restart the R session to try once more?

**Comment by isaactpetersen** [comment]
I got an error when trying to install using that command (`renv::install("rstudio/renv")`), but was able to install it from the repo by deactivating the `renv` project and installing it using the `{remotes}` package: (`remotes::install_github("rstudio/renv")`). Then, I re-activated the `renv` project, and updated the packages, and it worked successfully. Thanks very much for the fix!

**Comment by jhrcook** [comment]
> Thanks for taking the time to test. I think this should now be properly resolved in the development version of `renv`; are you able to test with e.g.
> 
> ```
> renv::install("rstudio/renv")
> ```
> 
> and then restart the R session to try once more?

I ran into the same error and installing the latest version from GitHub seems to have fixed it. Thank you for the quick response to this bug.

**Comment by kevinushey** [comment]
Thanks for the bug report. I've now published `renv 0.15.4` to CRAN, which should contain a fix for this issue. The new version should propagate to CRAN shortly.

If you need to update an `renv` project affected by this bug, you can use:

```
utils::install.packages("renv", type = "source")
renv::record("renv")
```

to install the latest version of `renv`, and then update the lockfile with the newly-installed copy of `renv`.

**Comment by grantmcdermott** [comment]
Same issue and same fix, i.e. 1) `renv::deactivate()`, 2) install newer version of renv, 3) `renv::activate()`, and finally 4) `renv::repair()`.
